workflow/automation/metadata/collect_metadata.py

```diff
@@ -0,0 +1,287 @@
+                _,
+                queued_time,
+                start_time,
+                end_time,
+                nodes,
+                cores,
+                memory,
+                WCT,
+            ) = db.execute(
+                "SELECT * from job_duration_log WHERE job_id=?",
+                (job_id,),
+            ).fetchone()
+            # Add runtime and cores to df
+            runtime = end_time - start_time
+            df.loc[rel_name, f"{proc_type_name}_runtime"] += runtime / 60
+            df = add_db_stat(df, rel_name, f"{proc_type_name}_cores", cores)
+            # Add to core hours
+            df.loc[rel_name, f"{proc_type_name}_core_hours"] += cores * runtime / 3600
+            df.loc[rel_name, f"Total_core_hours"] += cores * runtime / 3600
+            # Add to resubmits dict
+            if resub_counter.get(proc_type_name) is None:
+                resub_counter[proc_type_name] = 0
+            else:
+                resub_counter[proc_type_name] += 1
+    # Add resubmits to df
+    for k, v in resub_counter.items():
+        df.loc[rel_name, f"{k}_resubmits"] = v
+
+    # Extra Metadata
+    df.loc[rel_name, "IM_calc_components"] = len(params["ims"]["component"])
+    df.loc[rel_name, "EMOD3D_nt"] = params["sim_duration"] / params["dt"]
+    df.loc[rel_name, "HF_nt"] = params["sim_duration"] / params["dt"]
+    df.loc[rel_name, "BB_nt"] = params["sim_duration"] / params["dt"]
+    df.loc[rel_name, "IM_calc_BB_nt"] = params["sim_duration"] / params["dt"]
+    df.loc[rel_name, "AdvIM_models"] = len(params["advanced_IM"]["models"])
+    stations = get_stations(params["stat_file"])
+    df.loc[rel_name, "HF_n_stats"] = len(stations)
+    df.loc[rel_name, "BB_n_stats"] = len(stations)
+    df.loc[rel_name, "AdvIM_stations"] = len(stations)
+
+    # From IM calc csv
+    csv_ffp = f"{root_dir}/Runs/{fault_name}/{rel_name}/IM_calc/{rel_name}.csv"
+    try:
+        csv_columns = pd.read_csv(csv_ffp).columns.values
+        df.loc[rel_name, "IM_calc_pSA_count"] = len(
+            [True for column in csv_columns if "pSA" in column]
+        )
+        df.loc[rel_name, "IM_calc_FAS_count"] = len(
+            [True for column in csv_columns if "FAS" in column]
+        )
+    except FileNotFoundError:
+        print(f"Could not find file {csv_ffp} (Will not exist if IM_calc was not run)")
+    return df
+
+
+def main(root_dir: str, ch_count_type: str, out_ffp: str):
+    """
+    Gather metadata from each realisation and outputs to a csv
+    """
+    db = sql.connect(f"{root_dir}/slurm_mgmt.db")
+    rel_names = db.execute("SELECT DISTINCT run_name from state").fetchall()
+    df = pd.DataFrame(
+        columns=COLUMNS, data=np.zeros(shape=(len(rel_names), len(COLUMNS)))
+    )
+    df.index = [name_tuple[0] for name_tuple in rel_names]
+    df.index.name = "Rel_name"
+    for ix, name_tuple in enumerate(rel_names):
+        df = get_rel_info(df, name_tuple[0], root_dir, db, ch_count_type)
+    db.close()
```

Comment:
Should be using context managers?

Aside, it's probably better to move the sql to mgmtDB.py (it's called that or similar) and use the class.

I think that will probably solve my issue with the get_rel_info being a large function too.